enhancements/monitoring/20230131-tenant-impersonation.md

```diff
@@ -0,0 +1,38 @@
+# Tenant Impersonation
+
+## Summary:
+Tenant impersonation is a feature that will allow metrics written by privileged agents (e.g. the local agent) to optionally be forwarded to cortex under a different tenant id.
+
+## Table of contents
+* [Motivation](#Motivation)
+* [Implementation Details](#Implementation-Details)
+* [Scale and performance](#Scale-and-performance)
+* [Security](#Security)
+* [High availability](#High-availability)
+* [Testing](#Testing)
+
+## Motivation:
+In a multi-tenant environment, if a user only has access to a subset of clusters not including the central cluster, they will not be able to fetch the `opni_cluster_info` metric, as it is stored under the gateway's tenant id. Without `opni_cluster_info`, the user will not be able to see the names of the clusters they have access to inside Grafana.
+
+## Implementation Details
+
+The initial design to solve this problem was to have a "global tenant" that would be used to store metrics that all tenants would have access to. However, that design was determined not to be feasable to solve the problem at hand, because all tenants would have access to metrics that include descriptions of other tenants they may not have access to.
+
+This new design would allow a privileged agent to provide metric samples with specific labels identifying the tenant to impersonate for each sample. The gateway will be able to identify such samples using well-known metadata, patch labels as necessary, then switch the tenant id when pushing the samples to the Cortex distributor.
+
```

Comment:
Can we expand on this more, including diagrams of how you will envisage it working?